docs/guidelines/platforms/android/guideline_operable_android.md

```diff
@@ -282,6 +304,77 @@ The example given in the **Screenshot 4.** - **Avoid using ClickableSpan** and i
 
 ---
 
+### Heading and Labels (WCAG 2.4.6 - Level AA)
+
+Based on this guideline, users should clearly understand the purpose of the heading or label. It should be descriptive and give information about the content that comes next.
+
+> This guideline covers point *2.4.6 Headings and Labels - Level AA of the WCAG standard.*
+
+#### ✅ Success technique(s)
+
+When the user chooses to navigate between headings instead of between paragraphs or between word, make sure that sections on screen are defined as headings so that users can "skim" through them to locate the specific content they need.
+
+For Views, you can set the `android:accessibilityHeading` attribute to `true` for a view to be treated as a heading (requires minSdk >= 28). Alternatively, you can set `ViewCompat.setAccessibilityHeading(view, true)` for older versions.
+
+For Compose, you can use `heading()` semantics property to define a certain node as a heading.
+
+Example: setting a Text composable as a heading:
+```
+Text(
+    modifier = Modifier.semantics {
+        heading()
+    }
+```
+
+In general, try to make the headings and labels as descriptive as possible. Also, in addition to that, putting the most important information at the beginning of each heading helps users navigate through the content more easily.
+
+#### 🚫 Failures
+
+- Not providing a heading or label for the content that follows
+
+- Providing a missing or incorrect heading or label
+
+## Input modalities (WCAG 2.5)]
+
+*Make it easier for users to operate functionality through various inputs beyond keyboard.*
+
+### Label in Name (WCAG 2.5.3 - Level A)
+
+All user interface components that are defined as labels (that include text or images of text) contain the name, which is visible (presented visually).
+
+> This guideline covers point *2.5.3 Label in Name - Level A of the WCAG standard.*
+
+#### ✅ Success technique(s)
+
+A user should easily understand the information related to the focused label. To achieve this, the following steps should be taken:
+
+- content description should match the visible label name, or
```

Comment:
I've posted a note on the bottom which is a disclaimer for that:

> For most components like labels and buttons, accessibility service will handle things automatically. In case of a custom component, `contentDescription` label should be set manually.

However, this was more of a hint for custom components which might not handle content description properly on their own.

Reply:
I would highlight that note a bit more. Something  in the lines of :

**Important!** For most components like labels and buttons, accessibility service will handle things automatically. In case of a custom component, `contentDescription` label should be set manually.

Reply:
Updated